crates/revm/src/evm_impl.rs

```diff
@@ -318,6 +318,25 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> EVMImpl<'a, GSPEC, DB,
             });
         }
 
+        // Check initcode size limit on Shanghai spec.
+        if GSPEC::enabled(SHANGHAI) {
+            let max_initcode_size = self
+                .data
+                .env
+                .cfg
+                .limit_contract_code_size
+                .map(|limit| limit.saturating_mul(2))
+                .unwrap_or(MAX_INITCODE_SIZE);
+            if inputs.init_code.len() > max_initcode_size {
+                return Err(CreateResult {
+                    result: InstructionResult::CreateInitcodeSizeLimit,
+                    created_address: None,
+                    gas,
+                    return_value: Bytes::new(),
+                });
+            }
+        }
+
```

Comment:
This seems not needed, this check was moved inside `primitives/src/env`

Reply:
right, was wondering if this check was redundant as we were checking this before

Reply:
done in https://github.com/bluealloy/revm/pull/587/commits/345d69c8786f357cd88b68dab736d62b21ab03ed